docs/user-guides/federation.md

```diff
@@ -0,0 +1,180 @@
+**NOTE:** All examples used in this guide can be found under the [manifests
+directory](federation/manifests)
+
+### Create a Project to deploy MonitoringStack
+
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: federate-cmo
+```
+
+
+### Deploy Monitoring Stack
+
+
+Apply the following MonitoringStack
+
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: federate-cmo-ms
+  namespace: federate-cmo
+spec:
+  # 👇 Used to select the ServiceMonitor in the federate-cmo namespace
+  #    NOTE: there isn't a need for namespaceSelector
+  resourceSelector:
+    matchLabels:
+      monitoring.rhobs/stack: federate-cmo-ms
+
+  logLevel: info # 👈 use debug for verbose logs
+  retention: 3h
+
+  prometheusConfig:
+    replicas: 2  # 👈 ensures that at least one prometheus is running during upgrade
+
+  alertmanagerConfig:
+    disabled: true
+
+  resources: # 👈 ensure that you provide sufficient amount of resources
+    requests:
+      cpu: 500m
+      memory: 1Gi
+```
+
+
+### Grant Permission to Federate In-Cluster Prometheus
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: federate-cmo-ms-view
+  labels:
+    kubernetes.io/part-of: federate-cmo-ms
+    monitoring.rhobs/stack: federate-cmo-ms
+
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+subjects:
+- kind: ServiceAccount
+  # 👇 ServiceAccount used in the prometheus deployed by ObO.
+  #    SA name follows <monitoring stack name>-prometheus nomenclature
+  name: federate-cmo-ms-prometheus
+  namespace: federate-cmo
+```
+
+### Create ServiceMonitor for Federation
+
+```yaml
+apiVersion: monitoring.rhobs/v1
+kind: ServiceMonitor
+metadata:
+  name: federate-cmo-smon
+  namespace: federate-cmo
+  labels:
+    kubernetes.io/part-of: federate-cmo-ms
+    monitoring.rhobs/stack: federate-cmo-ms
+
+spec:
+  selector: # 👈 use the prometheus service to create a "dummy" target.
+    matchLabels:
+      app.kubernetes.io/managed-by: observability-operator
+      app.kubernetes.io/name: federate-cmo-ms-prometheus
+
+  endpoints:
+  - params:
+      'match[]': # 👈 scrape only required metrics from in-cluster prometheus
+        - '{__name__=~"container_cpu_.*", namespace="federate-cmo"}'
+        - '{__name__="container_memory_working_set_bytes", namespace="federate-cmo"}'
+
+    relabelings:
+    # 👇 relabel example
+    - targetLabel: source
+      replacement: my-openshift-cluster
+
+    # 👇 override the target's address by the prometheus-k8s service name.
+    - action: replace
+      targetLabel: __address__
+      replacement: prometheus-k8s.openshift-monitoring.svc:9091
+
+    # 👇 remove the default target labels as they aren't relevant in case of federation.
+    - action: labeldrop
+      regex: pod|namespace|service|endpoint|container
+
+    # 👇 30s interval creates 8 scrapes per minute
+    #    2 prometheus-k8s x 2 ms-prometheus x (60s/ 30s) = 8
```

Comment:
this should be 4 scrapes per minute: 1 prometheus-k8s target x 2 ms-prometheus x 2 scrapes/m

Reply:
Oh!! 
So the scrapes can potentially return different results depending on which prometheus responds?

Reply:
Thanks for pointing this out. Updated..